connectors/src/connectors/slack/auto_join_channel.ts

```diff
@@ -0,0 +1,81 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+
+import { joinChannel } from "@connectors/connectors/slack/lib/channels";
+import { getSlackClient } from "@connectors/connectors/slack/lib/slack_client";
+import {
+  SlackChannel,
+  SlackConfigurationModel,
+} from "@connectors/lib/models/slack";
+import type { Logger } from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+export function isChannelNameWhitelisted(
+  remoteChannelName: string,
+  whiteListedChannelPattern?: string | null
+): boolean {
+  if (!whiteListedChannelPattern || whiteListedChannelPattern === "") {
```

Comment:
The || ... === "" should not be needed right? The empty regex has a meaning and we might want to use it?